Tests/FluentAssertions.Specs/Xml/XDocumentAssertionSpecs.cs

```diff
@@ -1158,5 +1158,158 @@ public void When_asserting_a_document_has_an_element_with_a_null_xname_it_should
         }
 
         #endregion
+
+        #region HaveElement (with occurrence)
```

Comment:
🔧 We really should use nested classes for grouping related tests. This also allow you to have test names that don't need to repeat phrases like _When asserting_

Reply:
do you mean something like here?
https://github.com/fluentassertions/fluentassertions/blob/d6ad27985011df0bc5279bbaed182e5627658cdb/Tests/FluentAssertions.Specs/Collections/GenericCollectionAssertionOfStringSpecs.AllSatisfy.cs#L9-L15

Reply:
Should this reside in a seperate PR after this one?

Reply:
Hmm, maybe it's better to move that to a separate PR to not polute this one.